config/routes.rb

```diff
@@ -30,12 +30,11 @@
   get "*slug", slug: %r{help/.+}, to: "help#show", constraints: FormatRoutingConstraint.new('help_page')
 
   # Done pages
-  get "*slug", slug: %r{done/.+}, to: "root#publication"
-
-  # Transaction finished pages
-  constraints(slug: /(transaction-finished|driving-transaction-finished)/) do
-    get "/:slug.json"      => redirect("/api/%{slug}.json")
-    get "/:slug(.:format)" => "root#legacy_completed_transaction"
+  constraints FormatRoutingConstraint.new('completed_transaction') do
+    with_options(to: "completed_transaction#show") do
+      get "/:slug(.:format)", slug: /(transaction-finished|driving-transaction-finished)/ # Legacy
```

Comment:
So I've updated the routes file in the next commit, what do you think. It removes the duplication of those 2 specific slugs and just leaves it in the Controller. I'd say that there's not a lot of difference between the legacy and the normal case, other than the missing '/done' slug prefix and rendering a different partial. So the controller methods would be identical apart from the render call. The partials are actually also very similar, and the legacy ones should be removed once we do some house keeping. I've added [a story for that](https://trello.com/c/MsS4wrrf/479-convert-legacy-done-transaction-pages), what do you think?